apps/rig_inbound_gateway/lib/rig_inbound_gateway/api_proxy/router.ex

```diff
@@ -88,4 +92,37 @@ defmodule RigInboundGateway.ApiProxy.Router do
   end
 
   defp transform_req_headers(conn, _, _), do: conn
+
+  # ---
+
+  def add_forward_headers(conn) do
+    remote_ip = resolve_addr(conn.remote_ip)
+    host_ip = resolve_addr(@host)
+
+    forward_headers = [
+      {"Forwarded", "for=#{remote_ip};by=#{host_ip}"}
+    ]
+
+    updated_headers =
+      for(
+        {key, val} when key not in ["Forwarded"] <- conn.req_headers,
+        do: {key, val}
+      ) ++ forward_headers
```

Comment:
Hm just noticed the casing might be wrong - [according to the docs](https://hexdocs.pm/plug/Plug.Conn.html):

> req_headers - the request headers as a list, example: [{"content-type", "text/plain"}]. **Note all headers will be downcased** 

Also, the loop can now be written in a simpler way:

```suggestion
    updated_headers =
      conn.req_headers
      |> Enum.reject(fn {k, _} -> k === "forwarded" end)
      |> Enum.concat(forward_headers)
```

I think this also makes a bit clearer what the code is supposed to do.

Reply:
fixed